pkg/controller/client_utils.go

```diff
@@ -46,3 +46,37 @@ func preparePatchBytesforClusterStatus(oldCluster, newCluster *clusteroperator.C
 	}
 	return patchBytes, nil
 }
+
+func PatchMachineSetStatus(c clusteroperatorclientset.Interface, oldMachineSet, newMachineSet *clusteroperator.MachineSet) error {
+	logger := log.WithField("machineset", fmt.Sprintf("%s/%s", oldMachineSet.Namespace, oldMachineSet.Name))
+	patchBytes, err := preparePatchBytesForMachineSetStatus(oldMachineSet, newMachineSet)
+	if err != nil {
+		return err
+	}
+	logger.Debugf("about to patch machineset with %s", string(patchBytes))
```

Comment:
In manual testing this is showing up with some blank patch bytes:

```
$ oc logs cluster-operator-controller-manager-86f754b7d9-msbhq | grep "about to patch"
time="2018-01-29T14:37:12Z" level=debug msg="about to patch cluster with {\"status\":{\"conditions\":[{\"lastProbeTime\":\"2018-01-29T14:37:12Z\",\"lastTransitionTime\":\"2018-01-29T14:37:12Z\",\"message\":\"Job cluster-operator/job-infra-dgoodwin-cluster-t9fnm is running since \\u003cnil\\u003e. Pod completions: 0, failures: 0\",\"reason\":\"JobRunning\",\"status\":\"True\",\"type\":\"InfraProvisioning\"}]}}" cluster=cluster-operator/dgoodwin-cluster
time="2018-01-29T14:37:12Z" level=debug msg="about to patch cluster with {}" cluster=cluster-operator/dgoodwin-cluster
time="2018-01-29T14:37:13Z" level=debug msg="about to patch cluster with {}" cluster=cluster-operator/dgoodwin-cluster
time="2018-01-29T14:37:13Z" level=debug msg="about to patch cluster with {}" cluster=cluster-operator/dgoodwin-cluster
time="2018-01-29T14:37:13Z" level=debug msg="about to patch cluster with {}" cluster=cluster-operator/dgoodwin-cluster
time="2018-01-29T14:38:41Z" level=debug msg="about to patch cluster with {\"status\":{\"conditions\":[{\"lastProbeTime\":\"2018-01-29T14:38:41Z\",\"lastTransitionTime\":\"2018-01-29T14:38:41Z\",\"message\":\"Job cluster-operator/job-infra-dgoodwin-cluster-t9fnm completed at 2018-01-29 14:38:41 +0000 UTC\",\"reason\":\"JobCompleted\",\"status\":\"False\",\"type\":\"InfraProvisioning\"},{\"lastProbeTime\":\"2018-01-29T14:38:41Z\",\"lastTransitionTime\":\"2018-01-29T14:38:41Z\",\"message\":\"Job cluster-operator/job-infra-dgoodwin-cluster-t9fnm completed at 2018-01-29 14:38:41 +0000 UTC\",\"reason\":\"JobCompleted\",\"status\":\"True\",\"type\":\"InfraProvisioned\"}],\"provisioned\":true,\"provisionedJobGeneration\":1}}" cluster=cluster-operator/dgoodwin-cluster
time="2018-01-29T14:38:42Z" level=debug msg="about to patch machineset with {\"status\":{\"conditions\":[{\"lastProbeTime\":\"2018-01-29T14:38:42Z\",\"lastTransitionTime\":\"2018-01-29T14:38:42Z\",\"message\":\"Job cluster-operator/provision-machineset-dgoodwin-cluster-master-f7dlr-kj9wm is running since \\u003cnil\\u003e. Pod completions: 0, failures: 0\",\"reason\":\"JobRunning\",\"status\":\"True\",\"type\":\"HardwareProvisioning\"}]}}" machineset=cluster-operator/dgoodwin-cluster-master-f7dlr
time="2018-01-29T14:38:42Z" level=debug msg="about to patch machineset with {}" machineset=cluster-operator/dgoodwin-cluster-master-f7dlr
time="2018-01-29T14:38:42Z" level=debug msg="about to patch machineset with {}" machineset=cluster-operator/dgoodwin-cluster-master-f7dlr
```

Reply:
I would like to hold off on addressing this until a later PR, to keep this PR from growing with additional stuff.

Reply:
Works for me.